proposals/extensions.md

```diff
@@ -116,6 +109,254 @@ public implicit extension E for S
+
+## Open issue: allow attributes on extensions
+
+From WG discussion, we'd allow attributes on extensions types, using the AttributeTargets.Class target,
+since extensions are emitted as static classes.
+
+## Open issue: extension member lookup in usings
+
+We don't resolve implicit extension members in usings, due to cycles.  
+This needs to be investigated further to disentangle whether all cycles are implementation-specific
+or some are inherent to the language feature.
+
+## Open issue: need to specify scoping rules for `for UnderlyingType`
+
+Adjust scoping rules so that type parameters are in scope within the 'for':
+```csharp
+implicit extension E<T> for T { }
+```
+
+## Open issue: need to specify conversion rules
+
+```
+explicit extension R for U { } 
+R r = default;
+object o = r; // what conversion is that? if R doesn't have `object` as base type. What about interfaces?
+```
+
+Should allow conversion operators. Extension conversion is useful. 
+Example: from `int` to `string` (done by `StringExtension`).  
+But we should disallow user-defined conversions from/to underlying type 
+or inherited extensions, because a conversion already exists.  
+Conversion to interface still disallowed.  
+Could we make the conversion to be explicit identity conversion instead
+of implicit?  
+User-defined conversion should be allowed, except where it conflicts with a built-in
+conversion (such as with an underlying type).  
+
+## Open issue: need to specify constructor and operator members
+
+Do we want to allow constructors and `required` properties?  
+Do we want to allow operators?  
+
+## Open issue: need to specify type inference rules
+
+We need to review Method type inference: 7.5.2.9 Lower-bound interfaces, to see whether any updates are needed.  
+
+We want to allow the following and decide the inferred type:
+```csharp
+var c = new C();
+var e = (E)c; // E is an extension type
+M(c, e); // M<C> or M<E>?
+
+void M<T>(T t1, T t2) { }
+```
+
+## Open issue: naming convention?
+
+Should we have a naming convention like `Extension` suffixes? (`DataObjectExtension`)  
+
+## Open issue: need to specify pattern-based invocations and member access
+
+We need to scan through all pattern-based rules for known members to consider whether to include extension type members.
+  If extension methods were already included, then we should certainly include extension type methods.
+  Otherwise, we should consider it.
+  - `Deconstruct` in deconstruction and patterns
+  - `GetEnumerator`, `Current` and `MoveNext` in `foreach`
+  - CollectionBuilder in collection expressions
+  - `Add` in collection initializers
+  - `GetPinnableReference` in fixed statements
+  - `GetAwaiter` in `await` expressions
+  - `Dispose` in `using` statements
+
+## Open issue: need to disallow extensions within interfaces with variant type parameters  
+
+Will need to check with the WG as I don't recall the reasoning for this.
+
+## Open issue: need to specify the type erasure design
```

Comment:
What's the difference between this and "Open issue: need to specify type erasure" above?

Reply:
Merged now. Thanks